### PR TITLE
TLS Phase 1: SHA-512 (FIPS 180-4) + known-answer tests

### PR DESCRIPTION
**Pull request by kamrankhan78694**
## Subsystem
`tls` crypto — **SHA-512**, needed by Ed25519 (RFC 8032 hashes with SHA-512). Split out as its own focused primitive before Ed25519.

## What
- **`src/tls/sha512.{c,h}`** — streaming `sha512_init/update/final` + one-shot `sha512()`. Standard FIPS 180-4 (64-bit words, 80 rounds, 128-byte blocks, 128-bit big-endian length field). Mirrors the existing SHA-256 module's shape.

## Placement
Kept **TLS-gated** (`src/tls/`, behind `WEBLIB_TLS`) rather than in the shared `src/crypto/`, because — unlike SHA-256, which the JWT middleware uses on every target — no non-TLS subsystem needs SHA-512. So the default build carries no `sha512` symbol (`nm`-confirmed) and its `ctest` is unchanged. It can be promoted to `src/crypto/` later if a non-TLS caller appears.

## Tests
- FIPS/NIST vectors: `"abc"`, empty, and the **112-byte (two-block)** vector.
- A **1000-byte** message fed in **growing odd-size chunks** (1, 3, 7, 15, … spanning multiple blocks) to verify **streaming == one-shot**.

Vectors were confirmed against `hashlib.sha512` first. **Negative control:** a wrong init constant fails every vector.

## Validation
- TLS build (`-DWEBLIB_ENABLE_TLS=ON`): `ctest` **8/8**, warning-free; clean under **ASan/UBSan**.
- Default build (TLS OFF): **6/6** on the normal and ASan/UBSan trees — unchanged.

## Next
**Ed25519** (RFC 8032) sign + verify, using this SHA-512 and the curve field arithmetic — the final crypto primitive. After it, the crypto foundation for the handshake is complete.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by kamrankhan78694 — COMMENTED** [review]
## Independent review

Reviewed against FIPS 180-4, the padding/length edge cases, and scope. **No blocking issues.**

### Verified
- **Correct against the standard vectors**, including the two-block 112-byte NIST vector, and a streaming test that feeds 1000 bytes in growing odd chunks (1,3,7,15,…) so partial-buffer carry across block boundaries is exercised — it equals the one-shot digest. Confirmed against `hashlib` first; the NC (wrong init constant) fails everything.
- **Padding/length edge cases hold.** After the `0x80`, the "need another block" test is `index > SHA512_BLOCK_SIZE-16` (=112), so a message ending at 112..127 mod 128 correctly flushes an extra block; the `index == 112` boundary leaves exactly 16 bytes for the length with no spurious flush. The 128-bit length is encoded `bits_high = count>>61`, `bits_low = count<<3` (big-endian) — correct for byte counts up to 2^64.
- **Round core matches FIPS 180-4**: the four Σ/σ functions use the 64-bit rotation amounts (28/34/39, 14/18/41, 1/8/7, 19/61/6), 80 rounds, and the K[80] cube-root constants.
- **Scope**: gated behind `WEBLIB_TLS`; `nm` shows no `sha512` in the default lib; default `ctest` unchanged; ASan/UBSan clean.

### Note
Placed in `src/tls/` (TLS-gated) rather than `src/crypto/` because only Ed25519 needs it today. If a non-TLS caller ever wants SHA-512, promoting it to `src/crypto/` (WASM-safe, like SHA-256) is a clean follow-up.

_🤖 Reviewed by [Claude Code](https://claude.com/claude-code)_

File: src/tls/sha512.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+#include <stdint.h>
+
+#define SHA512_BLOCK_SIZE  128
+#define SHA512_DIGEST_SIZE 64
+
+typedef struct {
+    uint64_t state[8];
+    uint64_t count;      /* total message bytes (bit length is count*8) */
+    uint8_t  buffer[SHA512_BLOCK_SIZE];
+} sha512_ctx_t;
```
[the remark this answers is not in the document]
Comment (kamrankhan78694) [reply src/tls/sha512.h:30]: Fixed the documentation. The context intentionally uses a 64-bit *byte* counter, so the supported message length is 2^64-1 bytes; the SHA-512 128-bit *bit*-length field's higher bits are always 0 here (`bits_high = count>>61` captures the top 3 bits of the up-to-2^67-bit length exactly for any count a uint64 can hold). Reworded the struct comment — the prior "128-bit length" phrasing overstated it. A full 128-bit counter would only matter for a >16-exabyte message, which is physically unreachable and far beyond Ed25519's inputs, so I documented the ceiling rather than adding count_low/count_high.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: tests/test_tls_crypto.c [hunk 2]
```diff
@@ -362,6 +363,55 @@ static void test_hkdf(void) {
...
+/* SHA-512 (FIPS 180-4) known-answer tests, incl. a multi-block message and a
+ * streaming-equals-one-shot check. */
+static void test_sha512(void) {
+    uint8_t d[SHA512_DIGEST_SIZE];
+
+    sha512((const uint8_t *)"abc", 3, d);
+    check_hex("sha512 (\"abc\")", d, 64,
+              "ddaf35a193617abacc417349ae20413112e6fa4e89a97ea20a9eeee64b55d39a"
+              "2192992a274fc1a836ba3c23a3feebbd454d4423643ce80e2a9ac94fa54ca49f");
+
```
[the remark this answers is not in the document]
Comment (kamrankhan78694) [reply tests/test_tls_crypto.c:375]: Fair — the "negative control" is a dev-time verification I run on every crypto PR (temporarily break a constant, confirm the KAT fails, restore); it can't be a committed test because a committed test can't perturb the implementation. To give the suite a permanent guard, I added an input-sensitivity check (`sha512("abc") != sha512("abd")`) that catches a degenerate/constant-returning implementation. The NIST KATs remain the guard against a *wrong* algorithm.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
